### PR TITLE
Adds support for function calling with Anthropic models on Bedrock

### PR DESCRIPTION
**Pull request by bigbernnn**
Workaround for Bedrock support of function calling with Anthropic models. This change adds that `bind_tools` function to `ChatBedrock`.

```
from bedrock import ChatBedrock

chat = ChatBedrock(
    model_id=model_id,
    model_kwargs={"temperature": 0.1},
)

class GetWeather(BaseModel):
    """Get the current weather in a given location"""

    location: str = Field(..., description="The city and state, e.g. San Francisco, CA")

llm_with_tools = chat.bind_tools([GetWeather])
llm_with_tools

messages = [
    HumanMessage(
        content="what is the weather like in San Francisco"
    )
]
ai_msg = llm_with_tools.invoke(messages)
ai_msg
```

The workaround is implemented similarly to its equivalent directly using Anthropic prior to the feature currently in beta.

### CODE REVIEW

File: libs/aws/langchain_aws/chat_models/bedrock.py [hunk 1]
```diff
@@ -399,6 +428,43 @@ def get_token_ids(self, text: str) -> List[int]:
...
+            **kwargs: Any additional parameters to pass to the
+                :class:`~langchain.runnable.Runnable` constructor.
+        """
+        provider = self._get_provider()
+
+        if provider == "anthropic":
+            formatted_tools = [convert_to_anthropic_tool(tool) for tool in tools]
+            system_formatted_tools = get_system_message(formatted_tools)
+            self.set_system_prompt_with_tools(system_formatted_tools)
+            # llm_with_tools = self.bind(tools=system_formatted_tools, **kwargs)
```
[the remark this answers is not in the document]
Comment (bigbernnn) [reply libs/aws/langchain_aws/chat_models/bedrock.py]: comment removed

**Comment by bigbernnn** [comment]
> @bigbernnn Thanks for working on this change. Can you add integration tests or examples to test this.

Integration tests added for both generate and stream responses.

**Comment by zhongyu09** [comment]
Hi @bigbernnn, @3coins, from my side it seems the tool calling doesn't work for langgraph that relied on `tool_calls` in AIMessage. In line 399 of bedrock.py, when creating AIMessage, no `tool_calls` is passed to additional_kwargs.

**Comment by thiagotps** [comment]
Hi @3coins @bigbernnn. The current implementation also seems to ignore the `stop_reason` and `stop_sequence` fields returned by the Bedrock API when using the Claude 3 models.

**Comment by mhussar** [comment]
has any progress been made on this. There doesn't seem to be a standard result that can be parsed to receive the response required. it seems that we would have to rely on a hack using regular expressions

**Comment by vb-rob** [comment]
Try the new [ChatBedrockConverse](https://api.python.langchain.com/en/latest/chat_models/langchain_aws.chat_models.bedrock_converse.ChatBedrockConverse.html) model instead (still in beta). It supports tool calling, streaming, structured outputs, etc. Git repo [here](https://github.com/langchain-ai/langchain-aws/tree/main).

**Pull request merged**